internal/command/docker.go

```diff
@@ -79,7 +79,7 @@ func AppendEnv(args []string, environs map[string]string, quoteNeeded bool) []st
 		if quoteNeeded && strings.Contains(v, "=") {
 			// engine requires to double quote the env var when value contains
 			// the `=` char
-			env = fmt.Sprintf("%q", env)
+			env = fmt.Sprintf("%s=%q", k, v)
```

Comment:
I wonder if this could have a unit test? It potentially could have a big impact on other variable appending and we have seen a few bugs recently that suggest that not all envs encode the same...

Reply:
Not to sure what the test could be that doesn't look like just enforcing the same code as here, since it is following docker flags interface. Maybe we should exec docker and podman in our tests for that?

Reply:
It could be factored out to a helper that formats the arts and demonstrates that non-trivial inputs (spaces etc) create the right format of string.
Just a little sanity checking really as a simple "x=y" seems like a potential spot for glitches in env and cmd handling.

Reply:
Such a tests would enforce and lock implementation details, that are not necessarily correct. That's where I have trouble imagining a test without having to spin a docker and podman command line that is not locking implementation, but actually enforce that it work.

Reply:
I am totally lost - why does passing an env into a printf require a docker container?

Reply:
> > The printf is just trying to mimics docker/podman expectation.
> 
> Well, what is the expectation? It must be documented somewhere. Without tests we are constructing parameters with printf and hoping they are right. With tests we can document the expected behaviour - then if something is wrong we will see it in the expectations wrong and we can fix that.

We are talking exactly about the same thing, but where I see a problem you seems to a solution :-) Writing such a test enshrine a belief in my opinion, or as you say it document our expectation (there is no specification of what docker parameters are, there is documentation that are open to interpretation as with all integration problem). This doesn't mean that passing this unit test is correct (or not passing it is incorrect). It doesn't give us any confidence that for our next release things are any better. Also if we are wrong about our assumption, the test reduce velocity in fixing the issue as we now have two points to fix.

To me, this would be using the wrong tool, unit test, for the job as this is an integration job. If we want confidence in the result and no slow down in the future that the only test that make sense to write. So back to my point, the test that have value in my opinion for this problem require to bring docker and podman (because they do not necessarily perfectly match each other and their behavior can change over time) as a dependency during testing and they will have to download some small distribution so we can execute `export`. Is that ok to bring in? I am guessing I will have to protect those test behind a tag and only run it during ci on github. I think podman only work on mac node, but not sure will have to check what github action can run.

Reply:
I don't understand the argument here - why can we not test that a string is correctly escaped given certain inputs?

There is a comment on the previous line stating "_engine requires to double quote the env var when value contains the `=` char_" - so there is an expectation set somewhere. Here the behaviour has changed but the comment has not - so which is wrong? I don't think it is easy to tell without an illustration such as in a unit test.

Reply:
> I don't understand the argument here - why can we not test that a string is correctly escaped given certain inputs?

We can always write code, but the first question is `should we?`. In this case, writing a string comparison as unit test for an integration problem is not going to get any confidence toward the fact that we have a working code, instead it enforce a belief, which can be proven wrong at any point. This type of code is clear technical debt. Test can also be a carrier of technical debt.

> There is a comment on the previous line stating "_engine requires to double quote the env var when value contains the `=` char_" - so there is an expectation set somewhere. Here the behaviour has changed but the comment has not - so which is wrong? I don't think it is easy to tell without an illustration such as in a unit test.

The comment was actually correct, the code wasn't. If you look at the code, we are quoting the environment now, while before the value was being quoted.

Reply:
If that was the case then the comment "double quote the env var when value contains the `=` char" would surely have read "double quote the env value when value contains the `=` char", as the environment variable is in the KEY=VALUE format is it not?

Reply:
I agree with Andrew. It is better to verify that the code does what we think is right than not testing it at all.

Reply:
> If that was the case then the comment "double quote the env var when value contains the `=` char" would surely have read "double quote the env value when value contains the `=` char", as the environment variable is in the KEY=VALUE format is it not?

I see and understand that it might be confusing. In that case, any suggestion that would be less confusing for you?

Reply:
> It is better to verify that the code does what we think is right than not testing it at all.

I disagree on this statement. Tech debt for tests is worse than no tests. Never the less, it seems their is a consensus for having this kind of tests in, so I will oblige and add one.

Reply:
Test and comment change are in.